File: app/src/main/java/com/babestudios/companyinfouk/ui/charges/ChargesActivityView.java
package com.babestudios.companyinfouk.ui.charges;

import com.babestudios.companyinfouk.data.model.charges.Charges;

import net.grandcentrix.thirtyinch.TiView;
import net.grandcentrix.thirtyinch.callonmainthread.CallOnMainThread;

interface ChargesActivityView extends TiView {

	@CallOnMainThread
	void showProgress();

	@CallOnMainThread
	void hideProgress();

	@CallOnMainThread
	void showCharges(final Charges charges);

	@CallOnMainThread
	String getCompanyNumber();

	@CallOnMainThread
	void showNoCharges();

}

File: app/src/main/java/com/babestudios/companyinfouk/ui/charges/ChargesActivityView.kt
package com.babestudios.companyinfouk.ui.charges

import com.babestudios.companyinfouk.data.model.charges.Charges

import net.grandcentrix.thirtyinch.TiView
import net.grandcentrix.thirtyinch.callonmainthread.CallOnMainThread

interface ChargesActivityView : TiView {

	@get:CallOnMainThread
	val companyNumber: String

	@CallOnMainThread
	fun showProgress()

	@CallOnMainThread
	fun hideProgress()

	@CallOnMainThread
	fun showCharges(charges: Charges)

	@CallOnMainThread
	fun showNoCharges()

	fun showError()
}